apps/storybook-react/README.md

```diff
@@ -1,17 +1,24 @@
-# `@metamask/storybook-react`
```

Comment:
Updating storybook react readme to accurately describe the packages purpose and serve as an introduction page

<img width="1510" alt="Screenshot 2024-11-26 at 1 11 49 PM" src="https://github.com/user-attachments/assets/a82b96cf-483b-482a-8e47-ae80caef0879">